src/overlays/actors/ovl_Boss_Sst/z_boss_sst.c

```diff
@@ -1,15 +1,235 @@
+/*
+ * File: z_boss_sst.c
+ * Overlay: ovl_Boss_Sst
+ * Description: Bongo Bongo
+ */
+
 #include "z_boss_sst.h"
+#include "objects/object_sst/object_sst.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+#include "overlays/actors/ovl_Bg_Sst_Floor/z_bg_sst_floor.h"
 
 #define FLAGS 0x00000435
 
 #define THIS ((BossSst*)thisx)
 
+#define LEFT 0
+#define RIGHT 1
+#define OTHER_HAND(hand) ((BossSst*)hand->actor.child)
+#define HAND_STATE(hand) sHandState[hand->actor.params]
+
+typedef enum {
+    /*  0 */ HAND_WAIT,
+    /*  1 */ HAND_BEAT,
+    /*  2 */ HAND_RETREAT,
+    /*  3 */ HAND_SLAM,
+    /*  4 */ HAND_SWEEP,
+    /*  5 */ HAND_PUNCH,
+    /*  6 */ HAND_CLAP,
+    /*  7 */ HAND_GRAB,
+    /*  8 */ HAND_DAMAGED,
+    /*  9 */ HAND_FROZEN,
+    /* 10 */ HAND_BREAK_ICE,
+    /* 11 */ HAND_DEATH
+} BossSstHandState;
+
+typedef enum {
+    /* 0 */ BONGO_NULL,
+    /* 1 */ BONGO_ICE,
+    /* 2 */ BONGO_SHOCKWAVE,
+    /* 3 */ BONGO_SHADOW
+} BossSstEffectMode;
+
 void BossSst_Init(Actor* thisx, GlobalContext* globalCtx);
 void BossSst_Destroy(Actor* thisx, GlobalContext* globalCtx);
-void BossSst_Update(Actor* thisx, GlobalContext* globalCtx);
-void BossSst_Draw(Actor* thisx, GlobalContext* globalCtx);
+void BossSst_UpdateHand(Actor* thisx, GlobalContext* globalCtx);
+void BossSst_UpdateHead(Actor* thisx, GlobalContext* globalCtx);
+void BossSst_DrawHand(Actor* thisx, GlobalContext* globalCtx);
```

Comment:
just checking, all of these forward declares are needed right?

Reply:
Yes. Hence my question earlier of whether there's a reasonable way to move them to a different file.